### PR TITLE
Tweak: Plgs for all AI features before connect [ED-14096]

### PR DESCRIPTION
**Pull request by DennisNerush**

### CODE REVIEW

**Review by rotemee — CHANGES_REQUESTED** [review]

File: modules/ai/assets/js/editor/ai-behavior.js [hunk 1]
```diff
@@ -99,5 +122,32 @@ export default class AiBehavior extends Marionette.Behavior {
...
+			const rootBox = $button[ 0 ].getBoundingClientRect();
+			if ( ! rootBox || 0 === rootBox.width || 0 === rootBox.height ) {
+				return;
+			}
+
+			const controlType = this.view.model.get( 'type' );
+			const promotionTexts = this.getPromotionTexts( controlType );
+			if ( ! promotionTexts ) {
+				return;
+			}
```
[the remark this answers is not in the document]
Comment (DennisNerush) [reply modules/ai/assets/js/editor/ai-behavior.js:136]: I move the getPromotionTexts up but the button needs to stay here as it doesn't get the bounding at that point

File: modules/ai/assets/js/editor/components/ai-promotion-infotip-content.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	CardContent,
+	CardHeader,
+	Typography,
+} from '@elementor/ui';
+import { __ } from '@wordpress/i18n';
+
+const AiPromotionInfotipContent = ( props ) => {
+	return (
+		<Card sx={ { maxWidth: 345, width: 300 } }>
+			<CardHeader subheader="ELEMENTOR UI" />
```
[the remark this answers is not in the document]
Comment (DennisNerush) [reply modules/ai/assets/js/editor/components/ai-promotion-infotip-content.js:14]: AI :)

File: modules/ai/assets/js/editor/components/ai-promotion-infotip-wrapper.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+							focusOutListener.disable();
+							unmountAction();
+						} }
+						onClick={ () => {
+							markAsViewed();
+							focusOutListener.disable();
+							unmountAction();
+							anchor.trigger( 'click' );
+						} }
+					/> }
```
[the remark this answers is not in the document]
Comment (DennisNerush) [reply modules/ai/assets/js/editor/components/ai-promotion-infotip-wrapper.js:37]: what was changed?

[the remark this answers is not in the document]
Comment (rotemee) [reply modules/ai/assets/js/editor/components/ai-promotion-infotip-wrapper.js]: @DennisNerush - the rounded parenthesis:
`
content={ (
 // ...
) }
`

File: modules/ai/assets/js/editor/helpers/focus-out-listener.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+export function FocusOutListener( { children, listener, onFocusOut } ) {
+	const { indicatorRef, reset, remove, runAction } = listener;
+
+	React.useEffect( () => {
+		reset();
+
+		return () => {
+			remove();
+		};
```
Comment (matipojo) [inline modules/ai/assets/js/editor/helpers/focus-out-listener.js:11]:
```suggestion
		return remove;
```

**Review by matipojo — DISMISSED** [review]
Tests?

File: modules/ai/assets/js/editor/components/ai-promotion-infotip-wrapper.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	}
+
+	return (
+		<DirectionProvider rtl={ isRTL }>
+			<ThemeProvider colorScheme={ colorScheme }>
+				<FocusOutListener
+					listener={ focusOutListener }
+					onFocusOut={ () => unmountAction() }
+				>
+					<AiPromotionInfotip anchor={ anchor[ 0 ] }
```
Comment (rotemee) [inline modules/ai/assets/js/editor/components/ai-promotion-infotip-wrapper.js:22]:
```suggestion
					<AiPromotionInfotip anchor={ anchor }
```

File: modules/ai/assets/js/editor/ai-behavior.js [hunk 6]
```diff
@@ -99,5 +122,41 @@ export default class AiBehavior extends Marionette.Behavior {
...
+			if ( ! rootBox || 0 === rootBox.width || 0 === rootBox.height ) {
+				return;
+			}
+
+			const rootElement = document.createElement( 'div' );
+			document.body.append( rootElement );
+
+			const { unmount } = ReactUtils.render( (
+				<AiPromotionInfotipWrapper
+					anchor={ $button }
```
Comment (rotemee) [inline modules/ai/assets/js/editor/ai-behavior.js:149]:
```suggestion
					anchor={ $button[ 0 ] }
```

File: modules/ai/assets/js/editor/ai-behavior.js [hunk 7]
```diff
@@ -99,5 +122,41 @@ export default class AiBehavior extends Marionette.Behavior {
...
+			const rootBox = $button[ 0 ].getBoundingClientRect();
+			if ( ! rootBox || 0 === rootBox.width || 0 === rootBox.height ) {
+				return;
+			}
+
+			const rootElement = document.createElement( 'div' );
+			document.body.append( rootElement );
+
+			const { unmount } = ReactUtils.render( (
+				<AiPromotionInfotipWrapper
```
Comment (rotemee) [inline modules/ai/assets/js/editor/ai-behavior.js:148]: Seems that there is a certain offset with the info-tip pointer, have you seen it as well?

**Review by rotemee — CHANGES_REQUESTED** [review]

File: modules/ai/assets/js/editor/helpers/focus-out-listener.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+			<input
+				style={ {
+					width: 0,
+					height: 0,
+					padding: 0,
+					margin: 0,
+					outline: 0,
+					border: 0,
+					opacity: 0,
+				} }
```
Comment (rotemee) [inline modules/ai/assets/js/editor/helpers/focus-out-listener.js:27]:
```suggestion
					opacity: 0,
					position: 'fixed',
				} }
```
@DennisNerush  - to fix the vertical offset.

File: modules/ai/assets/js/editor/components/ai-promotion-infotip-content.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+	CardActions,
+	CardContent,
+	CardHeader,
+	Typography,
+} from '@elementor/ui';
+import { __ } from '@wordpress/i18n';
+
+const AiPromotionInfotipContent = ( props ) => {
+	return (
+		<Card sx={ { width: 300 } }>
```
Comment (rotemee) [inline modules/ai/assets/js/editor/components/ai-promotion-infotip-content.js:13]:
```suggestion
		<Card sx={ { width: 300 } } elevation={ 0 }>
```
@DennisNerush  - to fix the color difference between the box and the arrow of the tooltip.

**Review by rotemee — CHANGES_REQUESTED** [review]

**Review by rotemee — APPROVED** [review]

**Pull request merged**